clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2682,19 +2696,47 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
   }
   O << "};\n\n";
 
+  O << "// array representing kernel parameters used in all kernels defined in "
+       "the\n";
+  O << "// corresponding source\n";
+  O << "static constexpr\n";
+  O << "const bool param_omit_table[] = {\n";
+  O << "  //OMIT_TABLE_BEGIN\n";
+  for (unsigned I = 0; I < KernelDescs.size(); I++) {
+    auto &K = KernelDescs[I];
+    O << "  //--- " << K.Name << "\n";
+    O << "  ";
```

Comment:
As a future task (unrelated to this), we should probably be using O.indent(N) here instead of manually adding all the spaces.